### PR TITLE
Fix for #182: Last is always null

### PR DESCRIPTION
**Pull request by psytra**
While fiddling around with multiple yAxes, I've also stumbled over the fact that the "last" value is always null. There is already an existing Issue (#182). So here's a fix.

Vincent, if you already have a better solution for this, feel free to dump this PR.

 So here's the commit message..
```
Especially with asynchronous writing to RRD files, plot won't be able to
get the latest data. Instead the graph will end some seconds before. Now,
the "last" value is actually the last value shown in the graph.
```

The plot source a comment already mentions https://github.com/ziutek/rrd/pull/13. I didn't include it in the commit message, as even if this will be fixed some day, the problem with async backend writes will always be there.

### CODE REVIEW

File: pkg/plot/plot.go [hunk 1]
```diff
@@ -115,27 +115,27 @@ func (series *Series) Scale(factor Value) {
...
+		if !series.Plots[i].Value.IsNaN() {
+			current = series.Plots[i].Value
+			if current < min || min.IsNaN() {
+				min = series.Plots[i].Value
+			}
 
-		if series.Plots[i].Value > max {
-			max = series.Plots[i].Value
-		}
+			if current > max {
```
Comment (vbatoufflet) [inline pkg/plot/plot.go]: Maybe we need to add a `} else if …` to prevent even looking at the second if condition.

**Comment by vbatoufflet** [comment]
I have to specific issue with your patch. It seems good to me :+1: 

@falzm WDYT?

**Comment by falzm** [comment]
Good for me.

**Comment by psytra** [comment]
So this PR is pretty much finished for me. @vbatoufflet's rather cosmetic note about "else if" led me to the uninitialized max value, which was broken with only negative values. So I also enhanced the test code. I didn't split it into multiple PRs, as I think it can be considered as related.

**Comment by vbatoufflet** [comment]
Hi Matthias,

Indeed, I didn't catch the problem with negative values. Your PR is fine for me :+1:

**Pull request merged**